test/regression/RegressionTest.cc

```diff
@@ -29,84 +29,94 @@ void RegressionTest::run(const char* source, const char* triple,
-      std::make_unique<simeng::FlatMemoryInterface>(processMemory_,
-                                                    processMemorySize_);
-
+      std::make_unique<simeng::FlatMemoryInterface>(memory_);
   std::unique_ptr<simeng::FixedLatencyMemoryInterface> fixedLatencyDataMemory =
-      std::make_unique<simeng::FixedLatencyMemoryInterface>(
-          processMemory_, processMemorySize_, 4);
-  std::unique_ptr<simeng::MemoryInterface> dataMemory;
+      std::make_unique<simeng::FixedLatencyMemoryInterface>(memory_, 4);
 
-  // Create the OS kernel and the process
-  simeng::kernel::Linux kernel;
-  kernel.createProcess(*process_);
+  std::unique_ptr<simeng::MemoryInterface> dataMemory;
 
   // Populate the heap with initial data (specified by the test being run).
   ASSERT_LT(process_->getHeapStart() + initialHeapData_.size(),
             process_->getStackPointer());
-  std::copy(initialHeapData_.begin(), initialHeapData_.end(),
-            processMemory_ + process_->getHeapStart());
 
-  // Create the architecture
-  architecture_ = createArchitecture(kernel, config);
+  char* heapData = new char[initialHeapData_.size()];
+  std::copy(initialHeapData_.begin(), initialHeapData_.end(), heapData);
+  memory_->sendUntimedData(heapData, process_->getHeapStart(),
+                           initialHeapData_.size());
+  delete[] heapData;
 
   // Create a port allocator for an out-of-order core
   std::unique_ptr<simeng::pipeline::PortAllocator> portAllocator =
       createPortAllocator();
 
   // Create a branch predictor for a pipelined core
-  simeng::GenericPredictor predictor(config);
+  simeng::GenericPredictor predictor = simeng::GenericPredictor();
   // Create the core model
   switch (std::get<0>(GetParam())) {
     case EMULATION:
       core_ = std::make_unique<simeng::models::emulation::Core>(
-          instructionMemory, *flatDataMemory, entryPoint, processMemorySize_,
-          *architecture_);
+          instructionMemory, *flatDataMemory, *architecture_);
       dataMemory = std::move(flatDataMemory);
       break;
     case INORDER:
       core_ = std::make_unique<simeng::models::inorder::Core>(
-          instructionMemory, *flatDataMemory, processMemorySize_, entryPoint,
-          *architecture_, predictor);
+          instructionMemory, *flatDataMemory, *architecture_, predictor);
       dataMemory = std::move(flatDataMemory);
       break;
     case OUTOFORDER:
       core_ = std::make_unique<simeng::models::outoforder::Core>(
-          instructionMemory, *fixedLatencyDataMemory, processMemorySize_,
-          entryPoint, *architecture_, predictor, *portAllocator, config);
+          instructionMemory, *fixedLatencyDataMemory, *architecture_, predictor,
+          *portAllocator);
       dataMemory = std::move(fixedLatencyDataMemory);
       break;
   }
 
+  // Schedule Process on core
+  /** NOTE: Not using SimOS as core_ must be unique_ptr, but SimOS expects
```

Comment:
We had an offline conversation about why this may not be true. Core registration and SimOS ticking are required for upcoming developments.

Reply:
If this has been handled in upcoming developments, are there any changes needed here?

Reply:
Given that this isn't going straight into dev, probably not.

Reply:
Just double-checking I didn't make a mistake?

Reply:
I think you are correct yes. We will need to overhaul this function anyway in the future developments so I think can leave for now